modules/terraform/validate_test.go

```diff
@@ -35,3 +38,57 @@ func TestInitAndValidateWithError(t *testing.T) {
 	require.Error(t, err)
 	require.Contains(t, out, "Reference to undeclared input variable")
 }
+
+func TestReadModuleAndExampleSubDirsRejectsEmptyOpts(t *testing.T) {
+	opts := ValidationOptions{
+		RootDir:     "",
+		ExcludeDirs: []string{},
+	}
+
+	_, err := readModuleAndExampleSubDirs(opts)
+	require.Error(t, err)
+}
+
+func TestReadModuleAndExampleSubDirsExamples(t *testing.T) {
+	cwd, cwdErr := os.Getwd()
+	require.NoError(t, cwdErr)
+
+	opts := ValidationOptions{
+		RootDir:     filepath.Join(cwd, "../../"),
+		ExcludeDirs: []string{},
+	}
+
+	subDirs, err := readModuleAndExampleSubDirs(opts)
+	require.NoError(t, err)
+	// There are many valid Terraform modules in the root/examples directory of the Terratest project, so we should get back 0 results
+	require.Greater(t, len(subDirs), 0)
+}
+
+// Verify ExcludeDirs is working properly, by explicitly passing a list of two modules and two examples to exclude
+// and ensuring at the end that they do not appear in the returned slice of sub directories to validate
+func TestReadModuleAndExampleSubDirsWithResultsExclusion(t *testing.T) {
+
+	cwd, cwdErr := os.Getwd()
+	require.NoError(t, cwdErr)
+
+	projectRootDir := filepath.Join(cwd, "../..")
+
+	exclusions := []string{
+		filepath.Join(projectRootDir, "retry"),
+		filepath.Join(projectRootDir, "ssh"),
+		filepath.Join(projectRootDir, "terraform-packer-example"),
+		filepath.Join(projectRootDir, "terraform-hello-world-example"),
+	}
+
+	opts := ValidationOptions{
+		RootDir:     projectRootDir,
+		ExcludeDirs: exclusions,
+	}
+
+	subDirs, err := readModuleAndExampleSubDirs(opts)
+	require.NoError(t, err)
+	require.Greater(t, len(subDirs), 0)
+	for _, exclusion := range exclusions {
+		require.False(t, collections.ListContains(subDirs, exclusion))
```

Comment:
One issue with this test is that it doesn't confirm that the exclusions are included in the first place. Consider adding a step that calls the function WITHOUT exclusions, and then verifying the exclusion list is included in the results.

Otherwise, if we ever rename or remove examples, this test will still pass as a tautology.

Reply:
Fixed in 9519b48